app/controllers/assignments_controller.rb

```diff
@@ -561,14 +561,28 @@ def download_config_files
     FileUtils.rm_f(zip_path)
 
     Zip::File.open(zip_path, create: true) do |zipfile|
-      zipfile.get_output_stream('sample file') { |f| f.write 'Just a file' }
+      zipfile.get_output_stream("#{assignment.short_identifier}-properties.yml") { |f|
+        f.write assignment.assignment_properties_config.to_yaml }
     end
     send_file zip_path, filename: zip_name
   end
 
   # Uploads a zip file containing all the files specified in download_config_files
   # and modifies the assignment settings according to those files.
   def upload_config_files
+    begin
+      data = process_file_upload
```

Comment:
`process_file_upload` handles uploading csv or yml files. You are uploading a zip file so you have to handle this upload differently

Reply:
For example, look at the `UploadHelper#upload_files_helper` for an example of how to unzip an uploaded zip file